.github/workflows/bump-image-tags.yaml

```diff
@@ -20,15 +20,18 @@ jobs:
           # The Action can read multiple paths to images, but not multiple config files.
           - name: "pangeo-hubs common singleuser image"
             config_path: "config/clusters/pangeo-hubs/common.values.yaml"
-            values_paths: ".basehub.jupyterhub.singleuser.image"
+            # The regexpr attribute will ensure we only bump tags in the form YYYY.MM.DD
+            images_info: '[{"values_path": ".basehub.jupyterhub.singleuser.image", "regexpr": "[0-9]{4}.[0-9]{2}.[0-9]{2}"}]'
           - name: "m2lines common singleuser/PyTorch/Tensorflow"
             config_path: "config/clusters/m2lines/common.values.yaml"
+            # The regexpr attribute will ensure we only bump tags in the form YYYY.MM.DD
             # If the ordering of profileList changes, update the index in this expression
-            values_paths: ".basehub.jupyterhub.singleuser.image .basehub.jupyterhub.singleuser.profileList[4].kubespawner_override.image .basehub.jupyterhub.singleuser.profileList[5].kubespawner_override.image"
+            images_info: '[{"values_path"": ".basehub.jupyterhub.singleuser.image", "regexpr": "[0-9]{4}.[0-9]{2}.[0-9]{2}"}, {"values_path": ".basehub.jupyterhub.singleuser.profileList[4].kubespawner_override.image", "regexpr": "[0-9]{4}.[0-9]{2}.[0-9]{2}"}, {"values_path": ".basehub.jupyterhub.singleuser.profileList[5].kubespawner_override.image", "regexpr": "[0-9]{4}.[0-9]{2}.[0-9]{2}"}]'
           - name: "leap common singleuser/pangeo-ml-notebook"
             config_path: "config/clusters/leap/common.values.yaml"
+            # The regexpr attribute will ensure we only bump tags in the form YYYY.MM.DD
             # If the ordering of profileList changes, update the index in this expression
-            values_paths: ".basehub.jupyterhub.singleuser.image .basehub.jupyterhub.singleuser.profileList[4].kubespawner_override.image"
+            images_info: '[{"values_path": ".basehub.jupyterhub.singleuser.image", "regexpr": "[0-9]{4}.[0-9]{2}.[0-9]{2}"}, {"values_path": ".basehub.jupyterhub.singleuser.profileList[4].kubespawner_override.image", "regexpr": "[0-9]{4}.[0-9]{2}.[0-9]{2}"}]'
```

Comment:
Is this for dask images only, or is pangeo doing it as well? User defines the regular expression, and as for as I'm aware `[0-9]{4}.[0-9]{2}.[0-9]{2}` will pull the correct image tags for Pangeo. We can define another regex for dask images, but they are not being bumped in this config we have here.

Reply:
Maybe not, forgot they are different.

While we can adjust things later if found to be needed, it seems like a robustness win to not pin this soo tight.

My assumption is that the action does numerical sorting based on numerical identifiers though, not alphanumeric sorting.

Reply:
I guess {1,2} or {1-2} etc would be the change that could be possible.

Reply:
If this doesnt resonate with you go ahead, i don't want to block progress!

Reply:
This PR isn't currently about sorting, it's about filtering.

Pangeo publishes image tags that are either commits or dates, and we don't want the action to suggest a bump that replaces the tag with a commit-based one.

Reply:
> My assumption is that the action does numerical sorting based on numerical identifiers though, not alphanumeric sorting.

The algorithm sorts on the last_updated/last_modified field of the tag metadata, not its name. So we really are just talking about filtering out names that don't match the style we want.

Reply:
I considered the sorting aspect because the sorting after filtering is done could break down if the sorting wasn't done numerically and we included non-zero padded versions.

Please do go for a merge, this is fine as it is!